Observer/BalanceCustomerRegisterSuccessObserver.php

```diff
@@ -0,0 +1,79 @@
+<?php
+namespace Balancepay\Balancepay\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+use Balancepay\Balancepay\Model\Config;
+use Magento\Framework\Message\ManagerInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Customer\Model\Session;
+use Balancepay\Balancepay\Model\BalanceBuyer;
+
+class BalanceCustomerRegisterSuccessObserver implements ObserverInterface
+{
+    /**
+     * @var ManagerInterface
+     */
+    protected $_messageManager;
+
+    /**
+     * @var Config
+     */
+    private $balancepayConfig;
+
+    /**
+     * @var BalanceBuyer
+     */
+    private $balanceBuyer;
+
+    /**
+     * @var Session
+     */
+    private $customerSession;
+
+    /**
+     * Contructor
+     *
+     * @param ManagerInterface $messageManager
+     * @param Config $balancepayConfig
+     * @param BalanceBuyer $balanceBuyer
+     * @param Session $customerSession
+     */
+    public function __construct(
+        ManagerInterface $messageManager,
+        Config $balancepayConfig,
+        BalanceBuyer $balanceBuyer,
+        Session $customerSession
+    ) {
+        $this->_messageManager = $messageManager;
+        $this->balancepayConfig = $balancepayConfig;
+        $this->balanceBuyer = $balanceBuyer;
+        $this->customerSession = $customerSession;
+    }
+
+    /**
+     * Execute
+     *
+     * @param Observer $observer
+     * @return void
+     */
+    public function execute(Observer $observer)
+    {
+        $data = $observer['account_controller'];
+        try {
+            $paramData = $data->getRequest()->getParams();
+            if ($this->balancepayConfig->isActive() && !empty($paramData['email'])) {
```

Comment:
is there a use case where buyer email is missing? it's required for the Balance checkout

Reply:
No, email is required when creating a customer account in magento.

Reply:
This is just in case where other extension change the event data coming to our observer class.